### PR TITLE
Fix for #2414 - truncated copy/paste

### PR DESCRIPTION
**Pull request by psxpaul**
This is a fix for some of the issues raised in #2414. One thing to note - there was some question as to whether win32yank works for everyone. I did find a possible workaround for that, if it's still an issue:

```
const textToPaste = clipboard.readText()
const sanitizedTextLines = replaceAll(textToPaste, { "'": "''" })
await neovimInstance.command("let @+='" + sanitizedTextLines + "'")
```

I don't have a windows box to test this on, so perhaps we can add this code to `NeovimEditorCommands. pasteContents` later if needed.

### CODE REVIEW

File: browser/src/Input/KeyBindings.ts [hunk 1]
```diff
@@ -79,6 +80,7 @@ export const applyDefaultKeyBindings = (oni: Oni.Plugin.Api, config: Configurati
         input.bind("<c-,>", "oni.config.openConfigJs")
 
         if (config.getValue("editor.clipboard.enabled")) {
+            input.bind("<c-x>", "editor.clipboard.cut", isVisualMode)
```
Comment (badosu) [inline browser/src/Input/KeyBindings.ts]: `<c-x>` in visual mode decrements an integer value under the cursor

Comment (psxpaul) [reply browser/src/Input/KeyBindings.ts]: Thanks for the heads up, I'll remove this part

Comment (badosu) [reply browser/src/Input/KeyBindings.ts]: Well, you're not required to remove it, `<c-v>` is also overriden.

`<c-x>` makes some sense, perhaps `<C-c-x>`?

**Comment by Akin909** [comment]
@psxpaul seems the large paste test is repeatedly failing on windows 😭 we might need the solution in the snippet you pasted for windows

**Comment by CrossR** [comment]
I've not had a chance to test this yet, but I've set a reminder to do it tonight, so should be able to test it then. I'll also give it a test on my normal dev box, see if its all Windows or just the CI / base config versions of Oni.

**Comment by CrossR** [comment]
This is pretty odd....I ran the CI tests locally, it passed fine twice. I started Oni, pasted and copied a bunch, all good.

Then I tried it one last time, and now there is no clipboard provider registered, `:checkhealth` says its not hooked up...

**Comment by psxpaul** [comment]
Thanks for trying it out @CrossR. I got a Windows VM setup and managed to get the win32yank workaround working. I wasn't sure if this workaround should only be applied for windows machines, but the approach seems to work fine on my Mac and Linux machines.

**Comment by josemarluedke** [comment]
This PR might created a issue when pasting on command. Say I start a search with `/` and paste using `cmd + v`. The content will actually be pasted under the buffer that was focused instead of putting that content on the search.

**Pull request merged**